src/common/check_dataset_schema/script.py

```diff
@@ -0,0 +1,70 @@
+import anndata as ad
+import yaml
+import shutil
+import json
+
+
+## VIASH START
+# The following code has been auto-generated by Viash.
+par = {
+  'input': 'resources_test/common/pancreas/dataset.h5ad',
+  'schema': 'src/denoising/api/anndata_dataset.yaml',
+  'stop_on_error': 'false',
+  'copy_output': 'false',
+  'json': 'output/error.json',
+  'output': 'output/output.h5ad'
+}
+meta = {
+  'functionality_name': 'foo',
+
+}
+
+## VIASH END
+
+def check_structure (slot_info, adata_slot):
+  missing=[]
+  for obj in slot_info:
+      if obj['name'] not in adata_slot:
+         missing.append(obj['name'])
+
+  return missing
+
+def write_json(output):
+  if par['json'] is not None:
+    with open(par["json"], "w") as outf:
+      json.dump(output, outf, indent=2)
+
+
+print('Load data', flush=True)
+adata = ad.read_h5ad(par['input'])
+
+with open(par['schema'], 'r') as f:
+  data_struct = yaml.safe_load(f)
+
+
+def_slots = data_struct['info']['slots']
+
+out={
+  'exit_code' : 0,
+  'data_schema': 'ok',
+  'error': {
+    
+  }
+}
+
+for slot in def_slots:
+  check = check_structure(def_slots[slot], getattr(adata, slot))
+  if bool(check):
+    out['exit_code'] = 1
+    out['data_schema'] = 'not ok'
+    out['error'][slot] = check
+
+if par['stop_on_error'] == 'true':
+  write_json(out)
+  exit(out['exit_code'])
+
+if par['copy_output'] == 'true':
+  assert par['output'] is not None, 'No output defined'
+  write_json(out)
+  shutil.copyfile(par["input"], par["output"])
```

Comment:
Simplify code according to suggestion above

```suggestion
if par['json'] is not None:
  with open(par["checks"], "w") as f:
    json.dump(out, f, indent=2)

if par['output'] is not None:
  shutil.copyfile(par["input"], par["output"])
    
if par['stop_on_error']:
  exit(out['exit_code'])
```